ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/EeaSendRawTransaction.java

```diff
@@ -77,87 +72,100 @@ public String getName() {
 
   @Override
   public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
-    final String rawPrivateTransaction = requestContext.getRequiredParameter(0, String.class);
     final Object id = requestContext.getRequest().getId();
+    final String rawPrivateTransaction = requestContext.getRequiredParameter(0, String.class);
 
     try {
       final PrivateTransaction privateTransaction =
           PrivateTransaction.readFrom(RLP.input(Bytes.fromHexString(rawPrivateTransaction)));
 
       final String enclavePublicKey =
           enclavePublicKeyProvider.getEnclaveKey(requestContext.getUser());
-      final ValidationResult<TransactionInvalidReason> validationResult =
-          privacyController.validatePrivateTransaction(privateTransaction, enclavePublicKey);
-      if (!validationResult.isValid()) {
-        return new JsonRpcErrorResponse(
-            id, convertTransactionInvalidReason(validationResult.getInvalidReason()));
+
+      if (!privateTransaction.getPrivateFrom().equals(Bytes.fromBase64String(enclavePublicKey))) {
+        return new JsonRpcErrorResponse(id, PRIVATE_FROM_DOES_NOT_MATCH_ENCLAVE_PUBLIC_KEY);
       }
 
-      final String enclaveKey =
-          privacyController.sendTransaction(privateTransaction, enclavePublicKey);
-      final Transaction privacyMarkerTransaction;
-      if (privateTransaction.getPrivacyGroupId().isPresent()) {
-        PrivacyGroup privacyGroup = null;
-        try {
-          privacyGroup =
-              privacyController.retrievePrivacyGroup(
-                  privateTransaction.getPrivacyGroupId().get().toBase64String(),
-                  enclavePublicKeyProvider.getEnclaveKey(requestContext.getUser()));
-        } catch (final EnclaveClientException e) {
-          // it is an onchain group
+      PrivacyGroup privacyGroup = null;
+      final Optional<Bytes> maybePrivacyGroupId = privateTransaction.getPrivacyGroupId();
+      if (onchainPrivacyGroupsEnabled) {
+        if (!maybePrivacyGroupId.isPresent()) {
+          // on-chain private transaction does not contain privacy group ID
+          return new JsonRpcErrorResponse(
+              id, JsonRpcError.PRIVACY_GROUP_ID_NOT_AVAILABLE_WHILE_ON_CHAIN_PRIVACY_IS_ENABLED);
```

Comment:
We probably don't need this comment. The JsonRpcError name could be a bit simpler I think:
```
JsonRpcError.PRIVACY_GROUP_ID_NOT_AVAILABLE
```

Reply:
Done